src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Directives.cs

```diff
@@ -130,8 +132,11 @@ protected virtual void SectionDirective()
             {
                 editHandler.AutoCompleteString = "}";
                 Context.OnError(
-                    CurrentLocation,
-                    RazorResources.FormatParseError_Expected_X(Language.GetSample(CSharpSymbolType.RightBrace)),
+                    startingBraceLocation,
+                    RazorResources.FormatParseError_Expected_EndOfBlock_Before_EOF(
+                        SyntaxConstants.CSharp.SectionKeyword,
+                        Language.GetSample(CSharpSymbolType.RightBrace),
```

Comment:
lol, somewhat legacy, but it's just converting from Enum => string essentially. Didn't change this when we removed VB from Razor because it's still needed for the conversion.